src/api/sso/test/row-level-policies.test.js

```diff
@@ -0,0 +1,350 @@
+
+      expect(status).toBe(404);
+    });
+
+    it('should not allow deleting users', async () => {
+      const { status } = await supabase
+        .from('telescope_profiles')
+        .delete()
+        .match({
+          id: hash(hansLippershey.email),
+        });
+
+      expect(status).toBe(200);
+      await ensureUsers(users);
+    });
+  });
+
+  describe('"feeds" security policies', () => {
+    // Create feeds for HansLippershey and an unknown user
+    beforeEach(async () => {
+      await createTelescopeUsers([hansLippershey]);
+      await addFeeds(feedList);
+    });
+
+    afterEach(async () => {
+      await cleanFeeds(feedList);
+      await cleanupTelescopeUsers([hansLippershey]);
+    });
+
+    it('should allow reading the feeds list', async () => {
+      const { status, data: feeds } = await supabase.from('feeds').select('*');
+
+      expect(status).toBe(200);
+      expect(feeds.length).toBe(feedList.length);
+    });
+
+    it('should allow users to add feeds linked to their account', async () => {
+      const { status, data: feeds } = await supabase.from('feeds').insert(
+        {
+          url: 'http://dev.to/devto',
+          user_id: hash(hansLippershey.email),
+        },
+        { returning: 'representation' }
+      );
+
+      expect(status).toBe(201);
+      expect(feeds.length).toBe(1);
+      expect(feeds[0].url).toBe('http://dev.to/devto');
+    });
+
+    it('should not allow users to add feeds not linked to their account', async () => {
+      const { status } = await supabase.from('feeds').insert(
+        {
+          url: 'http://dev.to/devto',
+          wiki_author_name: 'devto',
+          user_id: hash(unknowUser.email),
+        },
+        { returning: 'minimal' }
+      );
+
+      expect(status).toBe(401);
+    });
+
+    it('should allow user to update their own feeds', async () => {
+      const { status, data: feeds } = await supabase
+        .from('feeds')
+        .update({ type: 'youtube' }, { returning: 'representation' })
+        .match({ url: hansLippershey.feeds[0] });
+
+      expect(status).toBe(200);
+      expect(feeds.length).toBe(1);
+      expect(feeds[0].type).toBe('youtube');
+    });
+
+    it("should not allow users to update others' feeds", async () => {
+      const { status } = await supabase
+        .from('feeds')
+        .update({ type: 'youtube' })
+        .match({ url: unknowUser.feeds[0] });
+
+      expect(status).toBe(404);
+    });
+
+    it('should allow users to delete their own feeds', async () => {
+      const { status } = await supabase.from('feeds').delete().match({
+        url: hansLippershey.feeds[0],
+      });
+
+      expect(status).toBe(200);
+      await ensureFeeds(feedList.slice(0, 1), false);
+    });
+
+    it("should not allow users to delete others' feeds", async () => {
+      const { status } = await supabase.from('feeds').delete().match({ url: unknowUser.feeds[0] });
+
+      expect(status).toBe(200);
```

Comment:
"should not" and 200 again?

Reply:
It returns 200 but no feeds are deleted. The line below ensures that.

Reply:
Do we get back any object with info about how many rows were affected, beyond the status?

Reply:
I'm using `count` returned now.